mssql_rawquery_block.py

```diff
@@ -0,0 +1,50 @@
+from nio.properties import Property, VersionProperty, StringProperty
+from nio.block.terminals import output
+from .mssql_base import MSSQLBase
+from nio.block.mixins.enrich.enrich_signals import EnrichSignals
+
+@output('results', label='Results')
+@output('no_results', label='No Results')
+class MSSQLRawQuery(EnrichSignals, MSSQLBase):
+
+    version = VersionProperty("1.0.0")
+    query = StringProperty(title='Parameterized Query (use ? for any user-supplied values)', default='SELECT * FROM table where id=?', order=10)
+    parameters = Property(title='Substitution Parameters (As List, In Order)', default='{{[]}}', order=11)

```

Comment:
I'm really tempted to use a `ListProperty(Property)` here.

Reply:
i guess the only reason that I can think of where you _would_ want to do this is if you are dynamically selecting the query based on a signal, which has different length parameter sets. To me, thats an *anti*-feature, though something that is _allowed_ based on the defined API. Something like:

```
query: "{{ 'select * from x where foo=?' if $foo is 'bar' else 'delete from x where bar=? and baz=?' }}"
parameters: "{{ [$wat] if $foo is 'bar' else [$bar, $baz] }}"
````

Butr *sweet jesus*, don't do that.

Reply:
Converted to ListProperty[Property]